apps/rush-lib/src/cli/RushCommandLineParser.ts

```diff
@@ -252,48 +211,60 @@ export class RushCommandLineParser extends CommandLineParser {
 
     // Register each custom command
     for (const command of commandLineConfiguration.commands) {
-      if (this.tryGetAction(command.name)) {
-        throw new Error(`${RushConstants.commandLineFilename} defines a command "${command.name}"`
-          + ` using a name that already exists`);
-      }
+      this._addCommandLineConfigAction(commandLineConfiguration, command);
+    }
+  }
 
-      this._validateCommandLineConfigCommand(command);
-
-      switch (command.commandKind) {
-        case 'bulk':
-          this.addAction(new BulkScriptAction({
-            actionName: command.name,
-            summary: command.summary,
-            documentation: command.description || command.summary,
-            safeForSimultaneousRushProcesses: command.safeForSimultaneousRushProcesses,
-
-            parser: this,
-            commandLineConfiguration: commandLineConfiguration,
-
-            enableParallelism: command.enableParallelism,
-            ignoreMissingScript: command.ignoreMissingScript || false,
-            ignoreDependencyOrder: command.ignoreDependencyOrder || false,
-            incremental: command.incremental || false,
-            allowWarningsInSuccessfulBuild: !!command.allowWarningsInSuccessfulBuild
-          }));
-          break;
-        case 'global':
-          this.addAction(new GlobalScriptAction({
-            actionName: command.name,
-            summary: command.summary,
-            documentation: command.description || command.summary,
-            safeForSimultaneousRushProcesses: command.safeForSimultaneousRushProcesses,
-
-            parser: this,
-            commandLineConfiguration: commandLineConfiguration,
-
-            shellCommand: command.shellCommand
-          }));
-          break;
-        default:
-          throw new Error(`${RushConstants.commandLineFilename} defines a command "${command!.name}"`
-            + ` using an unsupported command kind "${command!.commandKind}"`);
-      }
+  private _addCommandLineConfigAction(
+    commandLineConfiguration: CommandLineConfiguration | undefined,
+    command: CommandJson
+  ): void {
+    if (this.tryGetAction(command.name)) {
+      throw new Error(`${RushConstants.commandLineFilename} defines a command "${command.name}"`
+        + ` using a name that already exists`);
+    }
+
+    this._validateCommandLineConfigCommand(command);
+
+    switch (command.commandKind) {
+      case RushConstants.bulkCommandKind:
+        this.addAction(new BulkScriptAction({
+          actionName: command.name,
+
+          // To remain compatible with existing repos, `rebuild` defaults to calling the `build` command in each repo.
```

Comment:
This comment is misleading.

The `rush rebuild` and `rush build` command invoke the same NPM script because they share the same `package-deps-hash` state.  In other words:

```sh
# Rebuilds every project in the repo
$ rush rebuild
# Completes immediately without building anything,
# because all projects are already up to date
$ rush build
```

If they invoked different commands, then there it's difficult to prove that `rush build` can safely be skipped.